libs/concurrency/include/hpx/concurrency/cache_line_data.hpp

```diff
@@ -71,10 +71,7 @@ namespace hpx {
             typename NeedsPadding = typename detail::needs_padding<Data>::type>
         struct cache_aligned_data
         {
-            cache_aligned_data()
-              : data_{}
-            {
-            }
```

Comment:
I think we explicitly kept it like this, since some compilers go and initialize the padding otherwise: http://quick-bench.com/T-Z9iZ5Wn7E4WE2DoOqGEwVQPOE (i.e. GCC does it, clang doesn't seem to care either way). I suppose a comment would've been good.

Reply:
Excellent catch! I forgot about that. I'll revert this change.